flake.nix

```diff
@@ -5,42 +5,38 @@
   inputs.flake-utils.url = "github:numtide/flake-utils";
   inputs.flake-compat.url = "github:edolstra/flake-compat";
   inputs.flake-compat.flake = false;
+  inputs.solc-bin.url = "github:EspressoSystems/nix-solc-bin";
 
-  inputs.foundry.url = "github:shazow/foundry.nix/monthly";
-
-  outputs = { self, flake-utils, nixpkgs, foundry, ... }:
-    let
-      goVersion = 21; # Change this to update the whole stack
-      overlays = [
-        (final: prev: {
-          go = prev."go_1_${toString goVersion}";
-          # Overlaying nodejs here to ensure nodePackages use the desired
-          # version of nodejs.
-          nodejs = prev.nodejs-16_x;
-          pnpm = prev.nodePackages.pnpm;
-          yarn = prev.nodePackages.yarn;
-        })
-        foundry.overlay
-      ];
-    in
+  outputs = { flake-utils, nixpkgs, solc-bin, ... }:
     flake-utils.lib.eachDefaultSystem (system:
       let
+        goVersion = 21; # Change this to update the whole stack
+        overlays = [
+          # solc-bin.overlays.default
+          (final: prev: {
+            go = prev."go_1_${toString goVersion}";
+            # Overlaying nodejs here to ensure nodePackages use the desired
+            # version of nodejs.
+            nodejs = prev.nodejs-18_x;
```

Comment:
I had to update nixpkgs to get a new enough rust to build foundry but that doesn't contain nodejs-16 anymore. Not sure if updating to 18 breaks anything but we'll find out.

Reply:
Wait isn't rust not coming from nixpkgs but from oxalica/rust-overlay?

Reply:
I haven't looked into how to make buildRustPackage work with the oxalica overlay. It might be possible but since we don't need a rust toolchain in the dev shell this was easier. Upstream uses node v20 anyway so we might as well switch to that too.